test/e2e/prometheus_test.go

```diff
@@ -370,6 +371,95 @@ func TestBodySizeLimit(t *testing.T) {
 	}
 }
 
+func TestPrometheusPolicyHeaders(t *testing.T) {
+	const testAccount = "test-policy-headers"
+	const testNs = "test-policy-headers-ns"
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNs,
+			Labels: map[string]string{
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+	}
+	ns, err := f.KubeClient.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		foreground := metav1.DeletePropagationForeground
+		if err := f.KubeClient.CoreV1().Namespaces().Delete(ctx, testNs, metav1.DeleteOptions{PropagationPolicy: &foreground}); err != nil {
+			t.Logf("err deleting namespace %s: %v", testNs, err)
+		}
+	})
+
+	err = framework.Poll(2*time.Second, 10*time.Second, func() error {
+		_, err := f.CreateServiceAccount(testNs, testAccount)
+		return err
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = framework.Poll(2*time.Second, 10*time.Second, func() error {
+		_, err = f.CreateClusterRoleBinding(testNs, testAccount, "admin")
+		return err
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// The federate port (9092) is only exposed in-cluster so we need to use
+	// port forwarding to access kube-rbac-proxy.
+	host, cleanUp, err := f.ForwardPort(t, f.Ns, "prometheus-k8s", 9092)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanUp()
+
+	var client *framework.PrometheusClient
+	err = framework.Poll(5*time.Second, 5*time.Minute, func() error {
+		token, err := f.GetServiceAccountToken(testNs, testAccount)
+		if err != nil {
+			return err
+		}
+		client = framework.NewPrometheusClient(
+			host,
+			token,
+			&framework.QueryParameterInjector{
+				Name:  "namespace",
+				Value: testNs,
+			},
+		)
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = framework.Poll(5*time.Second, time.Minute, func() error {
+		resp, err := client.Do("GET", "/api/v1/labels", nil)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("expecting 200 status code, got %d (%q)", resp.StatusCode, resp.Body)
+		}
+
+		if resp.Header == nil {
+			return fmt.Errorf("expecting policy headers but is empty")
+		}
+
+		if resp.Header["Content-Security-Policy"][0] != "frame-ancestors 'none'" {
```

Comment:
(suggestion) simpler
```suggestion
		if resp.Header.Get("Content-Security-Policy") != "frame-ancestors 'none'" {
```